drake/systems/DrakeSystem.cpp

```diff
@@ -92,45 +133,129 @@ CascadeSystem::CascadeSystem(const DrakeSystemPtr& _sys1, const DrakeSystemPtr&
   state_frame = make_shared<MultiCoordinateFrame>("CascadeSystemState",initializer_list<CoordinateFramePtr>({continuous_state_frame,discrete_state_frame}));
 }
 
-DrakeSystem::VectorXs CascadeSystem::getX1(const VectorXs &x) const {
-  DrakeSystem::VectorXs x1(sys1->state_frame->getDim());
+VectorXd CascadeSystem::getX1(const VectorXd &x) const {
```

Comment:
I would pull CascadeSystem out of DrakeSystem.{h, cpp} and put it in its own file at this point.